4.x/typescript/Recenter/README.md

```diff
@@ -0,0 +1,273 @@
+
+The snippet above creates a `style` variable of type `Style`. The `textShadow` property updates to the specified string value upon interaction. In addition, three variables: `x`, `y`, and `scale` are set to the values of `this.state`.
+
+The UI is rendered based on the specified `div` element attributes:
+
+1.  First, the `bind` attribute is set to `this`, e.g. `bind={this}`.
+2.  Next, `class` is set to the `CSS.base` value, i.e. "recenter-tool".
+3.  The `styles` will reflect the `textShadow` property set a few lines prior.
+4.  The `onclick` event (note the lower case 'c'), is set to call the private `_defaultCenter` method once the widget is clicked.
+5.  Lastly, the widget UI itself will display the `x: <value>`, `y: <value>`, and `scale: <value>` of the current view.
+
+#### Export widget
+
+At the very end of the code page, add a line to [export](https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require) the object into an easily-consumable external module.
+
+```ts
+export default Recenter;
+```
+
+### 2. Compiling the TSX file
+
+Now that the widget's code is implemented, compile the _.tsx_ file to its underlying JavaScript implementation.
+
+In the command prompt, browse to the location of this sample directory and type`tsc`.
+
+This compiles any specified _.tsx_ files within the _tsconfig.json_'s files to their equivalent _.js_ files. You should now have a new _Recenter.js_ file generated in the same directory as its _.tsx_ file, in addition to a _Recenter.js.map_ [sourcemap](https://www.html5rocks.com/en/tutorials/developertools/sourcemaps/) file.
+
+### 3. Add the widget to the application
+
+Now that you generated the underlying _.js_ file for the widget, it can be added into your JavaScript application. In the same _Recenter_ directory, create an _index.html_ file.
+
+#### Add CSS
+
+The widget references the \`.recenter-tool' class.  Add a style element that references this class as seen below.
+
+```css
+.recenter-tool {
+  padding: 2em;
+  position: absolute;
+  top: 1em;
+  right: 1em;
+  color: #fff;
+  background: rgba(0, 0, 0, 0.5);
+  z-index: 999;
+}
+
+.recenter-tool>p {
+  margin: 0;
+}
+```
+
+#### Add the custom widget reference
+
+Once you've created the custom widget, you need to load it. This comes down to telling Dojo's module loader how to resolve the path for your widget which means mapping a module identifier to a file on your web server. On the SitePen blog, there's a [post](https://www.sitepen.com/blog/2013/06/20/dojo-faq-what-is-the-difference-packages-vs-paths-vs-aliases/) discussing the differences between aliases, paths and packages which may help alleviate any questions specific to this.
```

Comment:
Just curious if there's a more recent blog post on these topics? 2013 seems old.

Reply:
Oh yeah, that's old. I updated it to basic dojo sitepen doc